# 02 - Esperimenti Randomizzati

## Lo Standard Aureo

Nella sessione precedente, abbiamo visto perché e come l'associazione è diversa dalla causalità. Abbiamo anche visto cosa è richiesto per rendere l'associazione uguale alla causalità.

$
E[Y|T=1] - E[Y|T=0] = \underbrace{E[Y_1 - Y_0|T=1]}_{ATT} + \underbrace{\{ E[Y_0|T=1] - E[Y_0|T=0] \}}_{BIAS}
$

Per ricapitolare, l' **associazione diventa causalità se non c'è distorsione**. Non ci sarà distorsione se $E[Y_0|T=0]=E[Y_0|T=1]$. In parole semplici, l'associazione diventerà causalità **se i trattati e il controllo sono uguali o comparabili**, eccetto per il trattamento che ricevono. Oppure, in termini più tecnici, quando l'esito dei non trattati è uguale all'esito controfattuale dei trattati. Ricorda che questo esito controfattuale è l'esito del gruppo trattato se non avessero ricevuto il trattamento.

Penso che abbiamo fatto un buon lavoro nel spiegare come rendere l'associazione uguale alla causalità in termini matematici. Ma era solo in teoria. Ora, guardiamo il primo strumento che abbiamo per far sparire la distorsione: **Gli Esperimenti Randomizzati**. Gli esperimenti randomizzati **assegnano casualmente gli individui di una popolazione** a un trattamento o a un gruppo di controllo. La proporzione che riceve il trattamento **non deve necessariamente essere del 50%**. Potresti avere un esperimento in cui solo il 10% dei tuoi campioni riceve il trattamento.

La randomizzazione annichilisce la distorsione rendendo i risultati potenziali indipendenti dal trattamento.

$
(Y_0, Y_1) \perp\!\!\!\perp T
$

[n.d.t. il simbolo $\perp\!\!\!\perp$  indica l'**indipendenza statistica**. Dire che A e B sono indipendenti (A⊥⊥B) significa che conoscere il valore di una non ti dà assolutamente alcuna informazione sul valore dell'altra.

Invece (Y_0, Y_1) indica che stiamo considerando la **distribuzione congiunta** dei due risultati potenziali. Immagina che per ogni unità i (ogni scuola), ci sia un "pacchetto" informativo che contiene entrambi i destini possibili. La randomizzazione (T) deve essere indipendente dall'intero pacchetto.
]

Questo può essere confuso all'inizio (lo è stato per me). Ma non preoccuparti, mio coraggioso e sincero collega, lo spiegherò ulteriormente. Se l'esito è indipendente dal trattamento, ciò non implica anche che il trattamento non ha effetto? Beh, sì! Ma nota che non sto parlando degli esiti. Invece, sto parlando dei risultati **potenziali**. Il risultato potenziale è come l'esito **sarebbe stato** sotto trattamento ($Y_1$) o sotto controllo ($Y_0$). Nei trial randomizzati, **non** vogliamo che l'esito sia **indipendente** dal trattamento poiché pensiamo che **il trattamento causa** l'esito. Ma vogliamo che i risultati **potenziali** siano indipendenti dal trattamento.

![img](./data/img/rct/indep.png)

**Dire che i risultati potenziali sono indipendenti dal trattamento significa dire che sarebbero, in aspettativa, gli stessi nel gruppo trattato o nel gruppo di controllo**. In termini più semplici, significa che i gruppi trattati e di controllo sono comparabili. Oppure che conoscere l'assegnazione del trattamento non mi dà alcuna informazione su come l'esito era precedente al trattamento. Di conseguenza, $(Y_0, Y_1)\perp T$ significa che il trattamento è l'unica cosa che genera una differenza tra l'esito nei trattati e nel gruppo di controllo. 

Per vedere questo, nota che l'indipendenza implica precisamente che

$
E[Y_0|T=0]=E[Y_0|T=1]=E[Y_0]
$

Che, come abbiamo visto, fa sì che

$
E[Y|T=1] - E[Y|T=0] = E[Y_1 - Y_0]=ATE
$

Quindi, la **randomizzazione ci dà un modo per usare una semplice differenza nelle medie** tra il trattamento e il controllo e chiamare quello l'effetto del trattamento.

## In una Scuola Molto, Molto Lontana

Nel 2020, la pandemia di Coronavirus ha costretto le aziende ad adattarsi al distanziamento sociale. I servizi di consegna sono diventati diffusi e le grandi aziende si sono spostate verso una strategia di lavoro remoto. Con le scuole, non è stato diverso. Molte hanno avviato i loro stessi repository online di lezioni.

Quattro mesi dopo l'inizio della crisi, molti si chiedono se i cambiamenti introdotti potrebbero essere mantenuti. Non c'è dubbio che l'apprendimento online ha i suoi vantaggi. È più economico poiché può risparmiare su immobili e trasporti. Può anche essere più digitale, sfruttando contenuti di classe mondiale da tutto il mondo, non solo da un insieme fisso di insegnanti. Nonostante tutto ciò, abbiamo ancora bisogno di rispondere se l'apprendimento online ha un impatto negativo o positivo sulle prestazioni accademiche dello studente.

Un modo per rispondere è prendere studenti da scuole che offrono principalmente lezioni online e confrontarli con studenti da scuole che offrono lezioni in aule tradizionali. Come sappiamo ormai, questo non è il miglior approccio. Potrebbe accadere che le scuole online attraggono solo studenti ben disciplinati che ottengono risultati migliori della media anche se la classe fosse in presenza. In questo caso, avremmo una distorsione positiva, dove i trattati sono accademicamente migliori dei non trattati: $E[Y_0|T=1] > E[Y_0|T=0]$.

O al contrario, potrebbe accadere che le lezioni online siano più economiche e siano composte principalmente da studenti meno ricchi, che potrebbero dover lavorare oltre a studiare. In questo caso, questi studenti farebbero peggio rispetto a quelli delle scuole in presenza anche se seguissero lezioni in presenza. Se fosse questo il caso, avremmo una distorsione nella direzione opposta, dove i trattati sono accademicamente peggiori dei non trattati: $E[Y_0|T=1] < E[Y_0|T=0]$.

Quindi, sebbene potessimo fare semplici confronti, non sarebbe convincente. In un modo o nell'altro, non potremmo mai essere sicuri che non ci fosse una distorsione nascosta che mascherasse il nostro effetto causale.

![img](./data/img/rct/lurking_bias.png)

Per risolvere questo, abbiamo bisogno di rendere i trattati e i non trattati comparabili $E[Y_0|T=1] = E[Y_0|T=0]$. Un modo per forzare questo è assegnare casualmente le lezioni online e in presenza agli studenti. Se riuscissimo a farlo, il trattamento e i non trattati sarebbero, in media, uguali, ad eccezione del trattamento che ricevono.

Fortunatamente, alcuni economisti l'hanno fatto per noi. Hanno **randomizzato le lezioni** in modo che alcuni studenti fossero assegnati ad avere lezioni faccia a faccia, altri avessero solo lezioni online, e un terzo gruppo avesse un formato ibrido di lezioni sia online che faccia a faccia. Hanno raccolto dati su un esame standardizzato alla fine del semestre.

Ecco come si presenta il dataset:

In [2]:
import pandas as pd
import numpy as np

data = pd.read_csv("./data/online_classroom.csv")
print(data.shape)
data.head()

(323, 10)


,gender,asian,black,hawaiian,hispanic,unknown,white,format_ol,format_blended,falsexam
0,0,0.0,0.0,0.0,0.0,0.0,1.0,0,0.0,63.29997
1,1,0.0,0.0,0.0,0.0,0.0,1.0,0,0.0,79.96000
2,1,0.0,0.0,0.0,0.0,0.0,1.0,0,1.0,83.37000
3,1,0.0,0.0,0.0,0.0,0.0,1.0,0,1.0,90.01994
4,1,0.0,0.0,0.0,0.0,0.0,1.0,1,0.0,83.30000


Possiamo vedere che abbiamo **323 campioni**. Non è esattamente big data, ma qualcosa con cui possiamo lavorare. Per stimare l'effetto causale, possiamo semplicemente calcolare il punteggio medio per ogni gruppo di trattamento.



[n.d.t Face‑to‑face: lezione in presenza, Online: lezione a distanza, Blended: lezione mista]

In [3]:
(data
 .assign(class_format = np.select(
     [data["format_ol"].astype(bool), data["format_blended"].astype(bool)],
     ["online", "blended"],
     default="face_to_face"
 ))
 .groupby(["class_format"])
 .mean())

,gender,asian,black,hawaiian,hispanic,unknown,white,format_ol,format_blended,falsexam
class_format,,,,,,,,,,
blended,0.550459,0.217949,0.102564,0.025641,0.012821,0.012821,0.628205,0.0,1.0,77.093731
face_to_face,0.633333,0.202020,0.070707,0.000000,0.010101,0.000000,0.717172,0.0,0.0,78.547485
online,0.542553,0.228571,0.028571,0.014286,0.028571,0.000000,0.700000,1.0,0.0,73.635263


Esatto. È così semplice. Possiamo vedere che le lezioni faccia a faccia producono un punteggio medio di 78,54, mentre i corsi online producono un punteggio medio di 73,63. Non è una buona notizia per i sostenitori dell'apprendimento online. L'$ATE$ per una lezione online è quindi -4,91. Ciò significa che **le lezioni online causano agli studenti di ottenere prestazioni circa 5 punti inferiori**, in media. Punto. Non devi preoccuparti che i corsi online potrebbero avere studenti più poveri che non possono permettersi lezioni faccia a faccia o, per questo, non devi preoccuparti che gli studenti dei diversi trattamenti siano diversi in alcun modo diverso dal trattamento che hanno ricevuto. Per progettazione, l'**esperimento casuale è fatto per cancellare quelle differenze**.

Per questo motivo, un buon controllo di qualità, **sanity check, per vedere se la randomizzazione è stata fatta correttamente** (o se stai guardando i dati corretti) è verificare se i trattati sono uguali ai non trattati nelle variabili pre-trattamento. I nostri dati hanno informazioni su genere ed etnia per vedere se sono simili tra i gruppi. Possiamo dire che sembrano abbastanza simili per le variabili `gender`, `asian`, `hispanic` e `white`. La variabile `black`, tuttavia, sembra un po' diversa. Questo attira l'attenzione su cosa succede con un piccolo dataset. Anche sotto randomizzazione, potrebbe accadere che, per caso, un gruppo sia diverso da un altro. In campioni grandi, questa differenza tende a scomparire.

## L'Esperimento Ideale

Gli esperimenti randomizzati o **Randomized Controlled Trials (RCT)** sono il modo più affidabile per ottenere effetti causali. È una tecnica diretta e incredibilmente convincente. È così potente che la maggior parte dei paesi lo richiede per dimostrare l'efficacia di nuovi medicinali. Per fare un'analogia terribile, puoi pensare all'RCT come Aang, da Avatar: L'Ultimo Airbender, mentre altre tecniche sono più come Sokka. Sokka è cool e può fare alcuni trucchi ordinati qua e là, ma Aang può piegare i quattro elementi e connettersi con il mondo spirituale. Pensala così, se potessimo, l'RCT sarebbe tutto quello che faremmo per scoprire la causalità. **Un RCT ben progettato è il sogno di qualsiasi scienziato**.

![img](./data/img/rct/science_dream.png)

Sfortunatamente, tendono ad essere molto **costosi o semplicemente non etici**. A volte, semplicemente non possiamo controllare il meccanismo di assegnazione. Immagina di essere un medico che cerca di stimare l'effetto del fumo durante la gravidanza sul peso del bambino alla nascita. Non puoi semplicemente **costringere una porzione casuale di mamme a fumare durante la gravidanza**. Oppure supponiamo che lavori per una grande banca e devi stimare l'impatto della linea di credito sulla perdita di clienti. Sarebbe troppo costoso dare linee di credito casuali ai tuoi clienti. O che tu voglia capire l'impatto dell'aumento del salario minimo sulla disoccupazione. Non puoi semplicemente assegnare paesi ad avere un salario minimo o un altro. Afferri il concetto.

Vedremo più tardi **come abbassare il costo della randomizzazione utilizzando la randomizzazione condizionata**, ma non c'è niente che possiamo fare riguardo agli esperimenti non etici o fattibili. Tuttavia, ogni volta che affrontiamo domande causali, vale la pena pensare all'**esperimento ideale**. Chiediti sempre, se potessi, **quale sarebbe l'esperimento perfetto che realizzeresti per scoprire questo effetto causale?** Questo tende a fare luce su come possiamo scoprire l'effetto causale anche senza l'esperimento ideale.

## Il Meccanismo di Assegnazione

In un esperimento randomizzato, il meccanismo che assegna le unità a un trattamento o a un altro è, beh, casuale. Come vedremo più tardi, tutte le tecniche di inferenza causale cercheranno in qualche modo di identificare i meccanismi di assegnazione dei trattamenti. Quando sappiamo con certezza come si comporta questo meccanismo, l'inferenza causale sarà molto più sicura, anche se il meccanismo di assegnazione non è casuale.

Sfortunatamente, **il meccanismo di assegnazione non può essere scoperto semplicemente guardando i dati**. Ad esempio, se hai un dataset in cui l'istruzione superiore è correlata con la ricchezza, non puoi sapere con certezza quale ha causato quale guardando solo i dati. Dovrai usare la tua conoscenza su come funziona il mondo per argomentare a favore di un meccanismo di assegnazione plausibile: è il caso che le scuole educano le persone, rendendole più produttive e portandole a lavori meglio pagati. Oppure, se sei pessimista riguardo all'istruzione, puoi dire che le scuole non fanno nulla per aumentare la produttività, ed è solo una correlazione spuria perché solo le famiglie ricche possono permettersi che un bambino ottenga una laurea.

Nelle domande causali, di solito possiamo argomentare in entrambi i modi: che X causa Y, oppure che sia una terza variabile Z che causa sia X che Y, e quindi la correlazione tra X e Y è solo spuria. Per questo motivo, conoscere il meccanismo di assegnazione porta a una risposta causale molto più convincente. Questo è anche ciò che rende l'inferenza causale così entusiasmante. Mentre il **ML applicato di solito è solo premere alcuni pulsanti nell'ordine appropriato, l'inferenza causale applicata ti richiede di pensare seriamente al meccanismo** che genera quei dati.

## Idee Chiave

Abbiamo visto come gli esperimenti randomizzati siano il modo più semplice e più efficace per scoprire l'impatto causale. Lo fa rendendo i gruppi di trattamento e controllo comparabili. Sfortunatamente, non possiamo sempre fare esperimenti randomizzati, ma è comunque utile pensare a quale sarebbe l'esperimento ideale che faremmo se potessimo.

Qualcuno che ha familiarità con la statistica potrebbe stare protestando proprio ora perché **non ho esaminato la varianza della mia stima dell'effetto causale**. Come posso sapere che una diminuzione di 4,91 punti non è dovuta al caso? In altre parole, come posso sapere se la differenza è statisticamente significativa? E avrebbero ragione. Non preoccuparti. Ho intenzione di ripassare alcuni concetti statistici dopo.

## Riferimenti

Mi piace pensare a questo intero libro come un tributo a Joshua Angrist, Alberto Abadie e Christopher Walters per il loro fantastico corso di Econometria. La maggior parte delle idee qui sono tratte dai loro corsi presso l'American Economic Association. Guardarli è ciò che mi mantiene sano di mente durante questo difficile anno 2020.
* [Cross-Section Econometrics](https://www.aeaweb.org/conference/cont-ed/2017-webcasts)
* [Mastering Mostly Harmless Econometrics](https://www.aeaweb.org/conference/cont-ed/2020-webcasts)

Vorrei anche fare riferimento ai fantastici libri di Angrist. Mi hanno dimostrato che l'Econometria, o 'Metrics come la chiamano loro, non è solo estremamente utile ma anche profondamente divertente.

* [Mostly Harmless Econometrics](https://www.mostlyharmlesseconometrics.com/)
* [Mastering 'Metrics](https://www.masteringmetrics.com/)

Il mio ultimo riferimento è il libro di Miguel Hernan e Jamie Robins. È stato il mio fedele compagno nelle domande causali più spinose a cui ho dovuto rispondere.

* [Causal Inference Book](https://miguelhernan.org/whatifbook)

I dati utilizzati qui provengono da uno studio di Alpert, William T., Kenneth A. Couch, e Oskar R. Harmon. 2016. ["A Randomized Assessment of Online Learning"](https://www.aeaweb.org/articles?id=10.1257/aer.p20161057). American Economic Review, 106 (5): 378-82.

![img](./data/img/poetry.png)

## Contribuisci

Causal Inference for the Brave and True è un materiale open-source sull'inferenza causale, la statistica della scienza. Il suo obiettivo è essere accessibile sia economicamente che intellettualmente. Utilizza solo software libero basato su Python.
Se hai trovato questo libro prezioso e vuoi supportarlo, vai su [Patreon](https://www.patreon.com/causal_inference_for_the_brave_and_true). Se non sei pronto a contribuire finanziariamente, puoi anche aiutare correggendo errori di battitura, suggerendo modifiche o dando feedback su passaggi che non hai capito. Vai al repository del libro e [apri una issue](https://github.com/matheusfacure/python-causality-handbook/issues). Infine, se ti è piaciuto questo contenuto, condividilo con altri che potrebbero trovarlo utile e dagli una [stella su GitHub](https://github.com/matheusfacure/python-causality-handbook/stargazers).